incentivization.md

```diff
@@ -0,0 +1,342 @@
+The message sender generates a proof of enrollment in a membership set.
+Multiple proofs generated and revealed within one epoch lead to punishment.
+A valid RLN proof signifies that the message has been generated by a node with an active membership during a particular epoch.
+Note that a malicious node with a valid membership can generate messages but not broadcast them.
+Such messages would not be known to other nodes, but they would satisfy the RLN-based heuristic.
+We may later look into other ways for the client to check message relevance.
+
+## Privacy considerations
+
+Light protocols, in general, have weaker privacy properties than P2P protocols.
+In a client-server exchange, a client wants to selectively interact with the network.
+By doing so, it often reveals what it is interested in (e.g., subscribes to particular topics).
+
+A malicious Store server can spy on a client in the following ways:
+- track the topics the client is interested in;
+- analyze the periods of history interesting for the client;
+- analyze the timing of requests;
+- link requests made by the same client.
+
+Citing the [Store specification](https://rfc.vac.dev/spec/13/):
+> The main security consideration ... is that a querying node have to reveal their content filters of interest to the queried node, hence potentially compromising their privacy.
+
+### Service credentials
+
+Service credentials break the link between paying for the service and the service itself.
+Such scheme may be explored in the context of payment methods for higher user privacy.
+
+In a credential-based scheme:
+1. the client deposits funds into an on-chain pool;
+2. the client generates a credential that proves the transfer in zero-knowledge;
+3. the client sends the credential to the server;
+4. the server uses the credential to pull funds from the pool.
+
+Further reading: [one](https://forum.vac.dev/t/vac-sustainability-and-business-workshop/116), [two](https://github.com/vacp2p/research/issues/99), [three](https://github.com/vacp2p/research/issues/135).
+
+## Relation to long-term storage solutions
+
+Decentralized file storage networks, such as Codex, could (and perhaps should) be the backend for Store servers.
+Alternatives to Codex include IPFS, Filecoin, Sia, and Storj.
+We should explore this landscape and understand its relevance for Store i13n.
+
+## Should message senders pay?
+
+To ensure protocol sustainability, we should analyze its game theoretic properties.
+Note that there are in fact more than two party to the Store protocol:
+- the server;
+- the client;
+- the sender of the message.
+
+In particular, it is the _sender_ who imposes major costs on the server: the more messages the sender (or, indeed, multiple senders) broadcast, the higher are the Store server's storage costs.
+However, it is the Store client who pays for fetching messages.
+Is it fair / sustainable that the client pays for costs that the sender causes?
```

Comment:
This reminds me of gas fees on blockchain, in that what a user is paying for at the time of sending a tx is (primarily) the cost for nodes to perpetually store the history of this tx on the blockchain. In a sense, blockchains make the sender, not the "client" pay.

Reply:
I think this boils down to UX. IPFS is so popular because there is no friction for users. Adding a payment may discourage access, making it costly and a demanding process. However, if it is prepaid by the dev for their users and therefore, they can access it without paying, I think it can still work - but that is only one specific use scenario.

Reply:
Agreed. Perhaps there are two tiers: ephemeral messages (unpaid and unstored) and premium/pinned/permanent messages (paid and stored).

Reply:
> This reminds me of gas fees on blockchain

I think there is a key difference: on-chain execution is self-contained. Each node can deterministically conclude, based only on on-chain data, how much gas a transaction consumed. This is not the case with message relay.

> the cost for nodes to perpetually store the history

> two tiers: ephemeral messages (unpaid and unstored) and premium/pinned/permanent messages (paid and stored).

Related discussion: #32 .